src/components/SearchBar/index.js

```diff
@@ -1,9 +1,78 @@
-import AbstractSearchBar from "./AbstractSearchBar";
+import  React from 'react';
+import "react-filter-box/lib/react-filter-box.css";
+import "./index.css";
 
-export default class Search extends AbstractSearchBar {
+import ReactFilterBox from "react-filter-box";
 
-    handleSearch(newData) {
-        this.props.handleSearch(newData);
+import AutoCompleteHandler from './AutoCompleteHandler';
+import AdvancedResultProcessing from './AdvancedResultProcessing';
+
+import CheckCircle  from 'react-icons/lib/fa/check-circle';
+import TimeCircle  from 'react-icons/lib/fa/times-circle';
+import SearchIcon  from 'react-icons/lib/fa/search';
+  
+export default class SearchBar extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            data: [],
+            isOk: true,
+        }
+
+        // set props data
+        this.options = this.props.options;
```

Comment:
should this be done in componentWillMount? As well do you need these to be updated when any of the props have changed? What I mean do you need to add this in componentWillReceiveProps as well and update the autocompletehandler? What happens if the data is changed?